client/my-sites/stats/stats-post-anniversaries/index.jsx

```diff
@@ -0,0 +1,241 @@
+	render() {
+		const {
+			topPostsOnly,
+			oldestPostQuery,
+			postsByYearQueries,
+			postsByYear,
+			topPostsByYear,
+			siteId,
+			translate,
+			summary,
+			period,
+		} = this.props;
+
+		if ( ! siteId || PERIODS_ALLOWED.indexOf( period.period ) === -1 ) {
+			return null;
+		}
+
+		const allPosts = Array.prototype.concat( ...postsByYear );
+
+		let title = translate( 'Anniversaries', { comment: 'Title of the anniversaries module' } );
+		if ( topPostsOnly ) {
+			title = translate( 'Top %d anniversaries', {
+				args: MAX_POSTS_DISPLAYED,
+				comment: 'Title of the anniversaries module (when the maximum is reached)',
+			} );
+		}
+
+		return (
+			<div>
+				{ <QueryPosts siteId={ siteId } query={ oldestPostQuery } /> }
+				{ postsByYearQueries.map(
+					( query, i ) => <QueryPosts key={ i } siteId={ siteId } query={ query } />,
+				) }
+				{ ! allPosts.length
+					? null
+					: <div>
+							{ topPostsOnly
+								? allPosts.map(
+										( post, i ) => (
+											<QueryPostStats key={ i } siteId={ siteId } postId={ post.ID } />
+										),
+									)
+								: null }
+							{ ! summary &&
+								<SectionHeader label={ title } href={ summary ? null : this.getHref() } /> }
+							<Card className="stats-post-anniversaries__card stats-module">
+								{ summary &&
+									<div className="stats-post-anniversaries__title stats-section-title">
+										<h3>
+											{ translate( 'Anniversaries for %s', { args: this.getDateForDisplay() } ) }
+										</h3>
+									</div> }
+								<PostsList postsByYear={ topPostsByYear } summary={ summary } period={ period } />
+								{ topPostsOnly && <StatsModuleExpand href={ this.getHref() } /> }
+							</Card>
+						</div> }
+			</div>
+		);
+	}
+
+	getDateForDisplay() {
+		const { period, translate } = this.props;
+
+		// Ensure we have a moment instance here to work with.
+		const date = period.startOf.clone();
+
+		if ( period.period === 'week' ) {
+			return translate( '%(startDate)s - %(endDate)s', {
+				context: 'Date range to represent a week',
+				args: {
+					// LL is a date localized by momentjs
+					startDate: date.startOf( 'week' ).add( 1, 'd' ).format( 'LL' ),
+					endDate: date.endOf( 'week' ).add( 1, 'd' ).format( 'LL' ),
+				},
+			} );
+		}
+
+		if ( period.period === 'month' ) {
+			return date.format( 'MMMM YYYY' );
+		}
+
+		return date.format( 'LL' );
+	}
+
+	getHref() {
+		const { period, path, siteSlug } = this.props;
+
+		// Some modules do not have view all abilities
+		if ( period && path && siteSlug ) {
+			return (
+				`/stats/${ period.period }/${ path }/${ siteSlug }` +
+				`?startDate=${ period.startOf.format( 'YYYY-MM-DD' ) }`
+			);
+		}
+	}
+}
+
+const yearsAgoQuery = ( startDate, endDate, yearsAgo = 1 ) => ( {
+	after: startDate.clone().subtract( yearsAgo, 'years' ).format(),
+	before: endDate.clone().subtract( yearsAgo, 'years' ).format(),
+	status: 'publish',
+} );
+
+export default connect( () => {
```

Comment:
Having lots of logic in here indicates a code smell to me, and makes me think certain logic belongs in selectors?